apis/python/src/tiledbsoma/io/shaping.py

```diff
@@ -921,14 +1012,52 @@ def _get_new_ndarray_shape(
         )
 
 
+def _leaf_visitor_get_shapes(
+    item: Any,
+    *,
+    node_name: str,
+    nobs: int | None,
+    nvars: Dict[str, int] | None,
+    ms_name: str | None,
+    coll_name: str | None,
+    verbose: bool,
+    check_only: bool,
+    context: tiledbsoma.SOMATileDBContext | None,
+    output_handle: Printable | None,
+) -> dict[str, Any]:
+    retval: dict[str, Any] = {}
+    if isinstance(item, tiledbsoma.DataFrame):
+        retval["uri"] = item.uri
+        retval["type"] = "DataFrame"
+        retval["count"] = item.count
+        retval["non_empty_domain"] = item.non_empty_domain()
+        retval["domain"] = item.domain
+        retval["maxdomain"] = item.maxdomain
+        retval["upgraded"] = item.tiledbsoma_has_upgraded_domain
+
+    elif isinstance(item, tiledbsoma.SparseNDArray):
+        retval["uri"] = item.uri
+        retval["type"] = "SparseNDArray"
+        retval["non_empty_domain"] = item.non_empty_domain()
+        retval["shape"] = item.shape
+        retval["maxshape"] = item.maxshape
+        retval["upgraded"] = item.tiledbsoma_has_upgraded_shape
+
+    elif isinstance(item, tiledbsoma.DenseNDArray):
+        retval["uri"] = item.uri
+        retval["type"] = "DenseNDArray"
+        retval["shape"] = item.shape
+        retval["maxshape"] = item.maxshape
+        retval["upgraded"] = item.tiledbsoma_has_upgraded_shape
+
+    return retval
+
+
 def _check_statuses(dikt: dict[str, Any]) -> bool:
-    """For vistors which return status dicts, traverses the tree structure
-    and returns True only if all leaf-node statuses are True, else False.
-    """
     if "status" in dikt:
-        value = dikt["status"]
-        assert isinstance(value, bool)
-        return value
+        foo = dikt["status"]
+        assert isinstance(foo, bool)
+        return foo
```

Comment:
```suggestion
        ok = dikt["status"]
        assert isinstance(ok, bool)
        return ok
```

Reply:
Oops my bad I didn't mean to leave `foo` in there -- should have proofread more